### DISCUSSION TITLE
Part5의 Ch 01 파트 개요 및 실습 준비 파트에서 질문이 있습니다.

### DISCUSSION THREAD

**Discussion by shlee3048** [tedilabs/fastcampus-devops#71] [Q&A] [answered]
CH01_08. 테라폼 코드 이용하여 미리 준비된 AWS 실습환경 구성 강의
의 2분 30초와 같이
깃허브에서 코드를 다운로드 받아서 terraform init 후 terraform apply를 진행해보니 아래와 같은 에러가 나옵니다.

Error: error creating resource group: BadRequestException: Cannot create group: group already exists
│
│   with module.route_table__private.aws_resourcegroups_group.this[0],
│   on .terraform/modules/route_table__private/modules/route-table/resource-group.tf line 26, in resource "aws_resourcegroups_group" "this":
│   26: resource "aws_resourcegroups_group" "this" {
│
╵
╷
│ Error: error creating resource group: BadRequestException: Cannot create group: group already exists
│
│   with module.route_table__public.aws_resourcegroups_group.this[0],
│   on .terraform/modules/route_table__public/modules/route-table/resource-group.tf line 26, in resource "aws_resourcegroups_group" "this":
│   26: resource "aws_resourcegroups_group" "this" {
│
╵
╷
│ Error: error creating resource group: BadRequestException: Cannot create group: group already exists
│
│   with module.subnet_group["private"].aws_resourcegroups_group.this[0],
│   on .terraform/modules/subnet_group/modules/subnet-group/resource-group.tf line 26, in resource "aws_resourcegroups_group" "this":
│   26: resource "aws_resourcegroups_group" "this" {
│
╵
╷
│ Error: error creating resource group: BadRequestException: Cannot create group: group already exists
│
│   with module.subnet_group["public"].aws_resourcegroups_group.this[0],
│   on .terraform/modules/subnet_group/modules/subnet-group/resource-group.tf line 26, in resource "aws_resourcegroups_group" "this":
│   26: resource "aws_resourcegroups_group" "this" {
│
╵
╷
│ Error: error creating resource group: BadRequestException: Cannot create group: group already exists
│
│   with module.vpc.aws_resourcegroups_group.this[0],
│   on .terraform/modules/vpc/modules/vpc/resource-group.tf line 26, in resource "aws_resourcegroups_group" "this":
│   26: resource "aws_resourcegroups_group" "this" {

그런데 aws vpc 관리 콘솔에 들어가서 보면 vpc, 서브넷, 라우팅 테이블, 인터넷 게이트웨이가 모두 생성되어 있어서 계속해서 강의를 보며 진행했습니다.

그리고 이번에는 ec2 인스턴스를 만드는 6분27초와 같이 
ec2-instance 디렉토리로 디렉토리를 옮긴 후 terraform init과 apply를 하니 apne2-fastcampus-demo 인스턴스 하나만 생성이 되고 아래와 같은 에러가 발생했습니다.

null_resource.provisioner: Still creating... [4m50s elapsed]
null_resource.provisioner (remote-exec): Connecting to remote host via SSH...
null_resource.provisioner (remote-exec):   Host: 13.125.215.242
null_resource.provisioner (remote-exec):   User: ubuntu
null_resource.provisioner (remote-exec):   Password: false
null_resource.provisioner (remote-exec):   Private key: false
null_resource.provisioner (remote-exec):   Certificate: false
null_resource.provisioner (remote-exec):   SSH Agent: false
null_resource.provisioner (remote-exec):   Checking Host Key: false
null_resource.provisioner (remote-exec):   Target Platform: unix
null_resource.provisioner: Still creating... [5m0s elapsed]
╷
│ Error: remote-exec provisioner error
│
│   with null_resource.provisioner,
│   on ec2.tf line 57, in resource "null_resource" "provisioner":
│   57:   provisioner "remote-exec" {
│
│ timeout - last error: SSH authentication failed (ubuntu@13.125.215.242:22): ssh: handshake failed: ssh: unable to
│ authenticate, attempted methods [none], no supported methods remain

이 문제를 어떻게 해결해야 할까요?

**Comment by posquit0** [comment, marked as answer]
안녕하세요!
우선 혹시 이전 다른 파트(Ansible,Terraform, Packer 등)에서 이미 테라폼으로 네트워크를 구성하신 적이 있으신가요~?
도커 파트에서 실습 목적으로 사용되는 네트워크 테라폼 코드가 이전 파트에서 사용한 코드와 동일한 코드입니다! 만약 기존에 만드신 환경이 있다면, 그대로 사용을 하셔도 무방하며 혹은 기존 사용했던 디렉토리 상에서 `terraform destroy` 명령어로 모든 리소스를 안전하게 지우고 다시 시도해주시기 바랍니다. :)

`remote-exec`에서 발생하고 있는 오류는 SSH 연결 오류 이슈입니다. 이는 EC2 인스턴스에 SSH 연결을 위해 필요한 설정 진행이 필요로 한데요. SSH 연결 방법이 SSH Agent 이용, 사용자 계정 및 패스워드 인증, 비밀키 인증 등 다양한 방법이 있어서 강의 상에서 모든 경우를 다루기는 어려워 수강자의 선택에 맡긴 부분입니다...!

만약 사용자 계정 및 패스워드 인증이나 비밀키 인증을 사용하신다고 하면 아래 공식 문서 URL에서 해당하는 설정 값을 코드에 추가하여 시도해보시기 바랍니다.

- https://www.terraform.io/language/resources/provisioners/remote-exec

또 다른 도움 필요하시면 커멘트 남겨주세요~!!
감사합니다.

**Reply by shlee3048** [reply]
답변 정말 감사합니다
